app/src/main/java/org/jellyfin/androidtv/playback/nextup/NextUpButtons.kt

```diff
@@ -0,0 +1,66 @@
+package org.jellyfin.androidtv.playback.nextup
+
+import android.content.Context
+import android.os.CountDownTimer
+import android.util.AttributeSet
+import android.view.View
+import android.widget.FrameLayout
+import kotlinx.android.synthetic.main.fragment_next_up_buttons.view.*
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+
+class NextUpButtons(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0, defStyle: Int = 0) : FrameLayout(context, attrs, defStyleAttr, defStyle) {
+	constructor(context: Context, attrs: AttributeSet) : this(context, attrs, 0, 0)
+
+	private var countdownTimer: CountDownTimer? = null
+	private val view = View.inflate(context, R.layout.fragment_next_up_buttons, null)
+
+	init {
+		addView(view)
+		view.fragment_next_up_buttons_play_next.apply {
+			// Stop timer when unfocused
+			setOnFocusChangeListener { _, focused -> if (!focused) stopTimer() }
+
+			// Add initial focus
+			requestFocus()
+		}
+	}
+
+	fun startTimer() {
+		val duration = TvApp.getApplication().userPreferences.nextUpTimeout.toLong()
+
+		// Cancel current timer if one is already set
+		countdownTimer?.cancel()
+
+		// Create timer
+		countdownTimer = object : CountDownTimer(duration, 1) {
+			override fun onTick(millisUntilFinished: Long) {
+				fragment_next_up_buttons_play_next_progress.max = duration.toInt()
```

Comment:
This is probably a little bit nit-picky, but do we actually need to set max on every tick?

Reply:
So it says "every tick" but it won't actually call it every tick. It only calls the onTick function when it is able to. It's entirely based on the system performance.